lib/ui/painting.dart

```diff
@@ -552,6 +613,10 @@ class Paint {
         result.write(' hairline');
       if (strokeCap != StrokeCap.butt)
         result.write(' $strokeCap');
+      if (strokeJoin != StrokeJoin.miter)
+        result.write(' $strokeJoin');
+      if (strokeMiterLimit != 0.0)
+        result.write(' $strokeMiterLimit');
```

Comment:
just the number on its own will be weird. I would recommend:

```dart
      if (strokeJoin == StrokeJoin.miter) {
        if (strokeMiterLimit != 0.0)
          result.write(' $strokeJoin up to $strokeMiterLimit');
      } else {
        result.write(' $strokeJoin');
      }
```

...or some such.